src/nvim/msgpack_rpc/channel.c

```diff
@@ -715,6 +729,22 @@ const char *rpc_client_name(Channel *chan)
   return NULL;
 }
 
+const char *rpc_client_type(Channel *chan)
```

Comment:
Almost identical to `rpc_client_name`. Please avoid duplicate code. This can easily be generalized:

```c
const char *get_client_info(Channel *chan, const char *key)
  FUNC_ATTR_NONNULL_ALL
{
  if (!chan->is_rpc) {
    return NULL;
  }
  Dictionary info = chan->rpc.info;
  for (size_t i = 0; i < info.size; i++) {
    if (strequal(key, info.items[i].key.data)
        && info.items[i].value.type == kObjectTypeString) {
      return info.items[i].value.data.string.data;
    }
  }

  return NULL;
}
```

Reply:
I think `get_client_info_string_field` would be a better name, since there are non-string fields in `info`. What do you think?

Reply:
There are no callers using this for anything except string. A fully generalized approach can come later. Most important for now is to avoid duplicating code everywhere, that makes the future harder.